test/test_lib.py

```diff
@@ -169,19 +161,23 @@ def test_namespace(self):
         assert list(ops.lib._find_all_specs([tmpdir])) == []
 
 
-class TestLibParser(TestCase):
-    def _mkmod(self, name: str, content: typing.Optional[str] = None) -> ModuleSpec:
+class TestLibParser:
+    def _mkmod(
+        self,
+        request: pytest.FixtureRequest, name: str,
+        content: typing.Optional[str] = None
+    ) -> ModuleSpec:
         fd, fname = mkstemp(text=True)
-        self.addCleanup(os.unlink, fname)
+        request.addfinalizer(lambda: os.unlink(fname))
```

Comment:
Can we do this with the `tmp_path` fixture instead?

Reply:
Since using tmp_path here won't increase the length the arg list, I will do it.